diff -urN t/etc/init.d/boot u/etc/init.d/boot
--- t/etc/init.d/boot	Sun Sep  9 23:47:12 2001
+++ u/etc/init.d/boot	Mon Sep 10 13:06:41 2001
@@ -122,6 +122,8 @@
     fi
 fi
 
+rc_splash "early stop"
+
 #
 # s390 is 'like' serial console(better: no console ttys at all)
 # and we don't have access to the HW Clock
@@ -140,6 +142,8 @@
    FSCK_PROGRESSBAR=""
 fi
 
+rc_splash "fsck start"
+
 #
 # Find and activate volume groups (HM 1/1/1999)
 #
@@ -351,6 +355,7 @@
 echo "Activating remaining swap-devices in /etc/fstab..."
 swapon -a &> /dev/null
 rc_status -v1 -r
+rc_splash "fsck stop"
 
 # set and adjust the CMOS clock
 if test "$HWCLOCK_ACCESS" = "no" ; then
@@ -445,6 +450,8 @@
     rc_status -v -r
 fi
 
+rc_splash "bootnetwork start"
+
 #
 # initialize loopback device
 #
@@ -699,6 +706,7 @@
 # Let YaST2 finish its installation, if you installed with YaST2
 #
 if test -f /var/lib/YaST2/runme_at_boot ; then
+    rc_splash "YaST"
     exec 0<> $REDIRECT 1>&0 2>&0
     # if yast2 failed, this ensures proper system setup
     touch /var/lib/YaST2/run_suseconfig
diff -urN t/etc/init.d/halt u/etc/init.d/halt
--- t/etc/init.d/halt	Sat Sep  1 13:13:03 2001
+++ u/etc/init.d/halt	Mon Sep 10 13:06:41 2001
@@ -16,6 +16,8 @@
 trap "echo" SIGINT SIGSEGV SIGQUIT SIGTERM
 set +e
 
+rc_splash "reboot"
+
 case "$0" in
 	*halt)
 		message="The system will be halted immediately."
diff -urN t/etc/init.d/rc u/etc/init.d/rc
--- t/etc/init.d/rc	Mon Aug 27 17:08:07 2001
+++ u/etc/init.d/rc	Mon Sep 10 13:06:41 2001
@@ -115,6 +115,9 @@
     test $# -gt 1 && echo -e "$attn$runrc/: more than one link for service $service$norm"
     test -x "$1"  && continue
 
+    # send information to splash icon viewer.
+    rc_splash  "$i stop"
+
     # Log to /var/log/boot.msg
     blogger "$i stop"
 
@@ -144,6 +147,9 @@
     test $# -gt 1 && echo -e "$attn$prerc/: more than one link for service $service$norm"
     test -x "$1"  && continue
 
+    # send information to splash icon viewer.
+    rc_splash  "$i start"
+
     # Log to /var/log/boot.msg
     blogger "$i start"
 
@@ -175,6 +181,9 @@
 #    FAIL : 7 - program is not running (doing "start")
 #    OK   : 7 - program is not running (doing "stop")
 #
+
+rc_splash "master"
+
 echo -n "Master Resource Control: "
 echo -e "runlevel ${RUNLEVEL} has been ${stat}${extd}reached${norm}"
 if test -n "$failed" ; then
diff -urN t/etc/rc.status u/etc/rc.status
--- t/etc/rc.status	Tue Aug  7 16:55:36 2001
+++ u/etc/rc.status	Mon Sep 10 13:06:41 2001
@@ -31,6 +31,10 @@
 #    \015          carriage return (without newline)
 #
 
+# check whether splash screen animations are installed.
+_rc_splash=0
+test -x /bin/splash && _rc_splash=1
+
 if test -z "$LINES" -o -z "$COLUMNS" ; then
     eval `stty size 2>/dev/null | (read L C; \
 	  echo LINES=${L:-24} COLUMNS=${C:-80})`
@@ -196,4 +200,8 @@
 function rc_exit ()
 {
     exit $_rc_status_all
+}
+function rc_splash()
+{
+    test "$BOOT_SPLASH" != "no" && test "$_rc_splash" -eq 1 && /bin/splash "$1"
 }
