---
 etc/init.d/boot.swap |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/etc/init.d/boot.swap
+++ b/etc/init.d/boot.swap
@@ -96,7 +96,8 @@ case "$1" in
 	if [ ! -d "/sys/devices" ] || [ -d "/sys/power" ]; then
 	    check_swap_sig
 	fi
-	enable_ps3_vram_swap
+	# disable PS3 vram swap for now as it causes trouble with FW 2.50
+	# enable_ps3_vram_swap
 	echo "Activating remaining swap-devices in /etc/fstab..."
 	swapon -a &> /dev/null
 	rc_status -v1 -r
