--- etc/init.d/boot.localfs
+++ etc/init.d/boot.localfs
@@ -63,6 +63,8 @@
 	        # A return code of 2 or larger indicates failure.
 	        FSCK_RETURN=$?
 		test $FSCK_RETURN -lt 4
+                # XXX: > 4 is returned if filesystem is r/w mounted
+		test $FSCK_RETURN -gt 4 && FSCK_RETURN=0
 		rc_status -v1 -r
 		if test $FSCK_RETURN -gt 1 -a $FSCK_RETURN -lt 4 ; then
 		    # if appropriate, switch bootsplash to verbose 
